utils/zk-cycle-utils/tracer/src/main.rs

```diff
@@ -38,24 +38,67 @@ struct Args {
     #[arg(short, long)]
     /// Strip the hashes from the function name while printing
     strip_hashes: bool,
+
+    #[arg(short, long)]
+    /// Function name to target for getting stack counts
+    function_name: Option<String>,
+
+    #[arg(short, long)]
+    /// Exclude functions matching these patterns from display
+    /// usage: -e func1 -e func2 -e func3
+    exclude_view: Vec<String>,
 }
 
 fn strip_hash(name_with_hash: &str) -> String {
     let re = Regex::new(r"::h[0-9a-fA-F]+$").unwrap();
     re.replace(name_with_hash, "").to_string()
 }
 
-fn print_intruction_counts(count_vec: Vec<(&String, &usize)>, top_n: usize, strip_hashes: bool) {
+fn get_cycle_count(insn: &str) -> Result<usize, &'static str> {
+    match insn {
+        "LB" | "LH" | "LW" | "LBU" | "LHU" | "ADDI" | "SLLI" | "SLTI" | "SLTIU" |
+        "AUIPC" | "SB" | "SH" | "SW" | "ADD" | "SUB" | "SLL" | "SLT" | "SLTU" |
+        "XOR" | "SRL" | "SRA" | "OR" | "AND" | "MUL" | "MULH" | "MULSU" | "MULU" |
+        "LUI" | "BEQ" | "BNE" | "BLT" | "BGE" | "BLTU" | "BGEU" | "JALR" | "JAL" |
+        "ECALL" | "EBREAK" => Ok(1),
+
+        // Don't see this in the risc0 code base, but MUL, MULH, MULSU, and MULU all take 1 cycle,
+        // so going with that for MULHU as well.
+        "MULHU" => Ok(1),
+
+        "XORI" | "ORI" | "ANDI" | "SRLI" | "SRAI" | "DIV" | "DIVU" | "REM" | "REMU" => Ok(2),
+
+        _ => Err("Decode error"),
+    }
+}
```

Comment:
Can we add a comment here with a reference to the docs/upstream code where this is explained? If there is any

Reply:
will do!

Reply:
Nice, thank you!